.github/workflows/deploy-argo-application.yml

```diff
@@ -0,0 +1,102 @@
+---
+name: Deploy Argo Application
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+        description: environment to deploy
+        required: true
+      service_name:
+        type: string
+        description: service name
+        required: true
+    secrets:
+      ARGOCD_AUTH_TOKEN:
+        required: true
+
```

Comment:
This centralized workflow will eventually need to be updated to include the mute monitors step & required secrets along these lines: https://github.com/icariohealth/dhp-orchestration-service/pull/213/files#diff-407350ada8b249d936536719089497850852166be50650e3465ce34668c4631bR13

This will also require a change to the main workflow to pass these variables in:
https://github.com/icariohealth/dhp-orchestration-service/pull/213/files#diff-71cabc4177e41ea8f15a89eb65398fa8187739f68a2762706d84885cd8b2ddc3R45

To control for these breaking changes we'll probably want to cut a version / create a tag so that when you ask people to integrate this centralized definition they'll need to reference that version:

```
Deploy-Dev:
    uses: 'icariohealth/.github/.github/workflows/deploy-argo-application.yaml@v1'
```

This way:

- They can adopt the centralized deployment
- We can update the centralized definition for monitor suppression without breaking existing CD pipelines
- We can later notify teams to update the version and give them instructions for updating their main workflow when the are ready to adopt the monitoring changes.